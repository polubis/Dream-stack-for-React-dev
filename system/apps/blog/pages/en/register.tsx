import { RegisterView } from '../../views/register';

const RegisterPage = () => {
  return <RegisterView />;
};

export default RegisterPage;
