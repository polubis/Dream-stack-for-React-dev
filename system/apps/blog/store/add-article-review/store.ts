import { create } from 'zustand';
import type { AddArticleReviewStore } from './defs';
import { addArticleReviewForm } from './core';

const useAddArticleReviewStore = create<AddArticleReviewStore.State>(() => ({
  is: 'idle',
  form: addArticleReviewForm.init({ content: '' }),
}));

export { useAddArticleReviewStore, addArticleReviewForm };
