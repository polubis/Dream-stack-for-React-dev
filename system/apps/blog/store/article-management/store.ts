import { create } from 'zustand';
import type * as ArticleManagement from './defs';

const useArticleManagementStore = create<ArticleManagement.State>(() => ({
  is: 'idle',
}));

export { useArticleManagementStore };
