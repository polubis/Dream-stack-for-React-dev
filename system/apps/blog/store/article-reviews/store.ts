import { create } from 'zustand';
import type { ArticleReviewsStore } from './defs';

const useArticleReviewsStore = create<ArticleReviewsStore.State>(() => ({
  is: 'idle',
}));

export { useArticleReviewsStore };
