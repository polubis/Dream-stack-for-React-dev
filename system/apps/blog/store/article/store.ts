import { create } from 'zustand';
import { Article } from './defs';

const useArticleStore = create<Article.State>(() => ({ is: 'idle' }));

export { useArticleStore };
