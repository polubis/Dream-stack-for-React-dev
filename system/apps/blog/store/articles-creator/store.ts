import { create } from 'zustand';
import type { ArticlesCreator } from './defs';
import { creatorForm } from './form';

const useArticlesCreatorStore = create<ArticlesCreator.State>(() => ({
  is: 'idle',
  view: 'initial',
  form: creatorForm.init({
    title: '',
    description: '',
    tags: [],
    thumbnail: {
      file: null,
      preview: [],
    },
    content: '',
    lang: 'en',
    sendToReview: false,
  }),
}));

export { useArticlesCreatorStore };
