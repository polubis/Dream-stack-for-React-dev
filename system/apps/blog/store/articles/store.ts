import { create } from 'zustand';
import { articles_store_states } from './states';
import type { ArticlesStore } from './defs';

const useArticlesStore = create<ArticlesStore.State>(() =>
  articles_store_states.idle()
);

export { useArticlesStore };
