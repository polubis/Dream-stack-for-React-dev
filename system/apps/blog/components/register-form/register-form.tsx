import { Alert, Box, Button, Field, Font, Input } from '@system/figa-ui';
import { useRegisterStore } from '../../store/register';
import { get } from '@system/blog-selectors';

const RegisterForm = () => {
  const { key, form, setField, submit, error } = useRegisterStore();

  return (
    <Box
      spacing={[400, 150, 150, 150, 400, 400]}
      maxWidth="320px"
      margin="auto"
    >
      <Font variant="h6">Sign in into your account</Font>
      <Field label="Login*" hint="6 - 20 characters" error={form.errors.login}>
        <Input
          invalid={form.result.login}
          value={form.values.login}
          placeholder="Login*"
          autoFocus
          data-i={get('register-login-input')}
          onChange={(e) => setField('login', e.target.value)}
        />
      </Field>

      <Field label="Email*" error={form.errors.email}>
        <Input
          value={form.values.email}
          invalid={form.result.email}
          placeholder="Email*"
          type="email"
          data-i={get('register-email-input')}
          onChange={(e) => setField('email', e.target.value)}
        />
      </Field>

      <Field
        label="Password*"
        hint="6 - 20 characters"
        error={form.errors.password}
      >
        <Input
          value={form.values.password}
          invalid={form.result.password}
          type="password"
          placeholder="Password*"
          data-i={get('register-password-input')}
          onChange={(e) => setField('password', e.target.value)}
        />
      </Field>

      <Field
        label="Repeat your password*"
        hint="6 - 20 characters"
        error={form.errors.confirmPassword}
      >
        <Input
          value={form.values.confirmPassword}
          invalid={form.result.confirmPassword}
          placeholder="Repeat your password*"
          type="password"
          data-i={get('register-repeated-password-input')}
          onChange={(e) => setField('confirmPassword', e.target.value)}
        />
      </Field>

      <Box right>
        <Button
          data-i={get('register-confirm-btn')}
          onClick={submit}
          disabled={(form.touched || form.dirty) && form.invalid}
          loading={key === 'pending'}
        >
          Confirm
        </Button>
      </Box>

      {key === 'error' && (
        <Alert data-i={get('register-error-alert')} type="error">
          {error.message}
        </Alert>
      )}
      {key === 'ok' && (
        <Alert data-i={get('register-ok-alert')} type="ok">
          You are logged in!
        </Alert>
      )}
    </Box>
  );
};

export { RegisterForm };
