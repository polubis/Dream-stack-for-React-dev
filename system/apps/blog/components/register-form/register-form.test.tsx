import { render } from '@testing-library/react';
import { RegisterForm } from './register-form';
import { storeFixture } from '../../store/test-utils';
import { registerForm, useRegisterStore } from '../../store/register';

describe('User can register when: ', () => {
  it('validation messages are displayed', () => {
    const { restore } = storeFixture(useRegisterStore, {
      ...useRegisterStore.getState(),
      form: registerForm.set(
        registerForm.init({
          login: '',
          password: '',
          confirmPassword: '',
          email: '',
        })
      )({
        login: '23',
        password: 'dd',
        email: '',
        confirmPassword: 'D3',
      }),
    });

    const { asFragment } = render(<RegisterForm />);

    expect(asFragment()).toMatchSnapshot();

    restore();
  });
});
