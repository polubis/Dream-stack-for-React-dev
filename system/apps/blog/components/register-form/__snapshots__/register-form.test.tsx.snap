// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User can register when:  validation messages are displayed 1`] = `
<DocumentFragment>
  <div
    class="box column empty"
    style="margin: auto; max-width: 320px;"
  >
    <div
      class="box-item-wrapper"
      style="margin-bottom: 32px;"
    >
      <h6
        class="font h6 default"
      >
        Sign in into your account
      </h6>
    </div>
    <div
      class="box-item-wrapper"
      style="margin-bottom: 12px;"
    >
      <div
        class="box field invalid column empty"
      >
        <p
          class="font b2 default field-label"
        >
          Login*
        </p>
        <div
          class="filled invalid input"
        >
          <svg
            class="icon error-icon input-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 12C1.5 14.0767 2.11581 16.1068 3.26957 17.8335C4.42332 19.5602 6.0632 20.906 7.98182 21.7007C9.90045 22.4955 12.0116 22.7034 14.0484 22.2982C16.0852 21.8931 17.9562 20.8931 19.4246 19.4246C20.8931 17.9562 21.8931 16.0852 22.2982 14.0484C22.7034 12.0116 22.4955 9.90045 21.7007 7.98182C20.906 6.0632 19.5602 4.42332 17.8335 3.26957C16.1068 2.11581 14.0767 1.5 12 1.5C9.21523 1.5 6.54451 2.60625 4.57538 4.57538C2.60625 6.54451 1.5 9.21523 1.5 12ZM18.8625 17.8125L6.1875 5.1375C7.91298 3.70125 10.1129 2.96173 12.3556 3.0641C14.5983 3.16646 16.7218 4.10331 18.3092 5.69077C19.8967 7.27824 20.8335 9.40169 20.9359 11.6444C21.0383 13.8871 20.2988 16.087 18.8625 17.8125ZM6.18 18.87C4.36374 17.3276 3.23325 15.1279 3.03642 12.7533C2.83959 10.3787 3.59248 8.02286 5.13 6.2025L17.7975 18.87C16.1732 20.2425 14.1153 20.9955 11.9887 20.9955C9.86219 20.9955 7.80433 20.2425 6.18 18.87Z"
            />
          </svg>
          <input
            data-i="register-login-input"
            value="23"
          />
        </div>
        <p
          class="font b3 default field-hint italic"
        >
          minLength
        </p>
      </div>
    </div>
    <div
      class="box-item-wrapper"
      style="margin-bottom: 12px;"
    >
      <div
        class="box field invalid column empty"
      >
        <p
          class="font b2 default field-label"
        >
          Email*
        </p>
        <div
          class="filled invalid input"
        >
          <svg
            class="icon error-icon input-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 12C1.5 14.0767 2.11581 16.1068 3.26957 17.8335C4.42332 19.5602 6.0632 20.906 7.98182 21.7007C9.90045 22.4955 12.0116 22.7034 14.0484 22.2982C16.0852 21.8931 17.9562 20.8931 19.4246 19.4246C20.8931 17.9562 21.8931 16.0852 22.2982 14.0484C22.7034 12.0116 22.4955 9.90045 21.7007 7.98182C20.906 6.0632 19.5602 4.42332 17.8335 3.26957C16.1068 2.11581 14.0767 1.5 12 1.5C9.21523 1.5 6.54451 2.60625 4.57538 4.57538C2.60625 6.54451 1.5 9.21523 1.5 12ZM18.8625 17.8125L6.1875 5.1375C7.91298 3.70125 10.1129 2.96173 12.3556 3.0641C14.5983 3.16646 16.7218 4.10331 18.3092 5.69077C19.8967 7.27824 20.8335 9.40169 20.9359 11.6444C21.0383 13.8871 20.2988 16.087 18.8625 17.8125ZM6.18 18.87C4.36374 17.3276 3.23325 15.1279 3.03642 12.7533C2.83959 10.3787 3.59248 8.02286 5.13 6.2025L17.7975 18.87C16.1732 20.2425 14.1153 20.9955 11.9887 20.9955C9.86219 20.9955 7.80433 20.2425 6.18 18.87Z"
            />
          </svg>
          <input
            data-i="register-email-input"
            type="email"
            value=""
          />
        </div>
        <p
          class="font b3 default field-hint italic"
        >
          required
        </p>
      </div>
    </div>
    <div
      class="box-item-wrapper"
      style="margin-bottom: 12px;"
    >
      <div
        class="box field invalid column empty"
      >
        <p
          class="font b2 default field-label"
        >
          Password*
        </p>
        <div
          class="filled invalid input"
        >
          <svg
            class="icon error-icon input-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 12C1.5 14.0767 2.11581 16.1068 3.26957 17.8335C4.42332 19.5602 6.0632 20.906 7.98182 21.7007C9.90045 22.4955 12.0116 22.7034 14.0484 22.2982C16.0852 21.8931 17.9562 20.8931 19.4246 19.4246C20.8931 17.9562 21.8931 16.0852 22.2982 14.0484C22.7034 12.0116 22.4955 9.90045 21.7007 7.98182C20.906 6.0632 19.5602 4.42332 17.8335 3.26957C16.1068 2.11581 14.0767 1.5 12 1.5C9.21523 1.5 6.54451 2.60625 4.57538 4.57538C2.60625 6.54451 1.5 9.21523 1.5 12ZM18.8625 17.8125L6.1875 5.1375C7.91298 3.70125 10.1129 2.96173 12.3556 3.0641C14.5983 3.16646 16.7218 4.10331 18.3092 5.69077C19.8967 7.27824 20.8335 9.40169 20.9359 11.6444C21.0383 13.8871 20.2988 16.087 18.8625 17.8125ZM6.18 18.87C4.36374 17.3276 3.23325 15.1279 3.03642 12.7533C2.83959 10.3787 3.59248 8.02286 5.13 6.2025L17.7975 18.87C16.1732 20.2425 14.1153 20.9955 11.9887 20.9955C9.86219 20.9955 7.80433 20.2425 6.18 18.87Z"
            />
          </svg>
          <input
            data-i="register-password-input"
            type="password"
            value="dd"
          />
        </div>
        <p
          class="font b3 default field-hint italic"
        >
          minLength
        </p>
      </div>
    </div>
    <div
      class="box-item-wrapper"
      style="margin-bottom: 32px;"
    >
      <div
        class="box field invalid column empty"
      >
        <p
          class="font b2 default field-label"
        >
          Repeat your password*
        </p>
        <div
          class="filled invalid input"
        >
          <svg
            class="icon error-icon input-icon"
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.5 12C1.5 14.0767 2.11581 16.1068 3.26957 17.8335C4.42332 19.5602 6.0632 20.906 7.98182 21.7007C9.90045 22.4955 12.0116 22.7034 14.0484 22.2982C16.0852 21.8931 17.9562 20.8931 19.4246 19.4246C20.8931 17.9562 21.8931 16.0852 22.2982 14.0484C22.7034 12.0116 22.4955 9.90045 21.7007 7.98182C20.906 6.0632 19.5602 4.42332 17.8335 3.26957C16.1068 2.11581 14.0767 1.5 12 1.5C9.21523 1.5 6.54451 2.60625 4.57538 4.57538C2.60625 6.54451 1.5 9.21523 1.5 12ZM18.8625 17.8125L6.1875 5.1375C7.91298 3.70125 10.1129 2.96173 12.3556 3.0641C14.5983 3.16646 16.7218 4.10331 18.3092 5.69077C19.8967 7.27824 20.8335 9.40169 20.9359 11.6444C21.0383 13.8871 20.2988 16.087 18.8625 17.8125ZM6.18 18.87C4.36374 17.3276 3.23325 15.1279 3.03642 12.7533C2.83959 10.3787 3.59248 8.02286 5.13 6.2025L17.7975 18.87C16.1732 20.2425 14.1153 20.9955 11.9887 20.9955C9.86219 20.9955 7.80433 20.2425 6.18 18.87Z"
            />
          </svg>
          <input
            data-i="register-repeated-password-input"
            type="password"
            value="D3"
          />
        </div>
        <p
          class="font b3 default field-hint italic"
        >
          minLength
        </p>
      </div>
    </div>
    <div
      class="box-item-wrapper"
      style="margin-bottom: 32px;"
    >
      <div
        class="box column empty right"
      >
        <button
          class="button size-3 rectangle filled primary"
          data-i="register-confirm-btn"
          disabled=""
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
